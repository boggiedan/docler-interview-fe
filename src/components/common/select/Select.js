import React, { useEffect } from "react";
import PropTypes from "prop-types";

const Select = ({ options, selectedOption, onSelect }) => {
  const changeHandler = event => onSelect(event.target.value);

  const renderOptions = option => (
    <option key={option.id} value={option.id}>
      {option.value}
    </option>
  );

  return (
    <select value={selectedOption || ""} onChange={changeHandler}>
      {options.map(renderOptions)}
    </select>
  );
};

Select.propTypes = {
  options: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      value: PropTypes.string.value
    }).isRequired
  ).isRequired,

  selectedOption: PropTypes.number,

  onSelect: PropTypes.func.isRequired
};

Select.defaultProps = {
  options: []
};

export default Select;
