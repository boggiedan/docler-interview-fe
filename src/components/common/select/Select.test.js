import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Select from "./Select";

const options = [
  { id: 1, value: "toto" },
  { id: 2, value: "titi" },
  { id: 3, value: "tutu" }
];

const onSelect = jest.fn();

describe("Select test", () => {
  let component;

  beforeEach(() => {
    component = shallow(<Select options={options} onSelect={onSelect} />);
  });

  it("should match snapshot", () => {
    expect(toJson(component)).toMatchSnapshot();
  });

  it("should call onSelect on value change", () => {
    component.find("select").simulate("change", { target: {} });
    expect(onSelect).toHaveBeenCalledTimes(1);
  });

  it("should call onSelect with correct value", () => {
    component.find("select").simulate("change", { target: { value: 1 } });
    expect(onSelect).toHaveBeenCalledWith(1);
  });
});
