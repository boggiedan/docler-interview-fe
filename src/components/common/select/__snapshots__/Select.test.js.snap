// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Select test should match snapshot 1`] = `
<select
  onChange={[Function]}
  value=""
>
  <option
    key="1"
    value={1}
  >
    toto
  </option>
  <option
    key="2"
    value={2}
  >
    titi
  </option>
  <option
    key="3"
    value={3}
  >
    tutu
  </option>
</select>
`;
